lib/elixir/lib/access.ex

```diff
@@ -360,6 +382,11 @@ defmodule Access do
   end
 
   def pop(nil, key) do
+    pop_raise(key)
+  end
+
+  @doc false
+  def pop_raise(key) do
```

Comment:
The thrid option would be to move the pop_raise/1 function to `Kernel.Utils`.
If that doesn't work, I'm thinking of  leaving the as it was and deal with the Dialyzer error